### PR TITLE
Remove the overlay logout step

### LINKED ISSUE #4554: Logout-Overlay entfernen
Mit dem neu eingeführten `recently touched` Menü, welcher ausgecheckte Dokumente als Badge-Counter darstellt, ist die LogoutOverlay Funktionalität, überflüssig. Ein weitere Warnung das noch ausgecheckte Dokumente vorhanden sind, eher störend, da doppelt. 

Deshalb soll das `logout entfernt werden und der Logout direkt erfolgen.

### PR DESCRIPTION
**Pull request by tarnap**
The URL is computed, a view is therefore required.
The view returns the logout URL which is then used by the logout action.

logout

Resolves #4554

### CODE REVIEW

File: opengever/document/browser/logout_overlay.py [hunk 1]
```diff
@@ -21,19 +19,5 @@ def redirect_url(self):
...
-    def update(self):
-        self.items = self.get_checkedout_documents()
-
     def __call__(self):
-        self.update()
-        if not self.items:
-            return "empty:%s" % self.redirect_url
-
-        return ViewPageTemplateFile('templates/logout_overlay.pt')(self)
+        return self.request.RESPONSE.redirect(self.redirect_url)
```
Comment (phgross) [inline opengever/document/browser/logout_overlay.py]: There should no longer be a need for a separate BrowserView, there should be a direct link and the multi-client logout should be registered for the action itself.

**Comment by tarnap** [comment]
@phgross: Could you re-review this PR?

File: opengever/base/tests/test_logout_overlay.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from opengever.setup.casauth import install_cas_auth_plugin
+from opengever.testing import FunctionalTestCase
+from plone import api
+import transaction
+
+
+class TestLogoutWithoutCASAuth(FunctionalTestCase):
+
+    @browsing
+    def test_user_is_redirected_to_logout(self, browser):
```
Comment (phgross) [inline opengever/base/tests/test_logout_overlay.py]: `test_url_is_plone_logout`

File: opengever/base/tests/test_logout_overlay.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        transaction.commit()
+
+    def tearDown(self):
+        super(TestLogoutWithCASAuth, self).tearDown()
+        acl_users = api.portal.get_tool('acl_users')
+        acl_users.plugins.removePluginById('cas_auth')
+        transaction.commit()
+
+    @browsing
+    def test_user_is_redirected_to_cas_server_logout(self, browser):
```
Comment (phgross) [inline opengever/base/tests/test_logout_overlay.py]: It's not redirected

`test_url_is_cas_server_logout`

File: opengever/base/tests/test_logout_overlay.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+from ftw.testbrowser import browsing
+from ftw.testbrowser.exceptions import HTTPClientError
+from opengever.setup.casauth import install_cas_auth_plugin
+from opengever.testing import FunctionalTestCase
+from plone import api
+import transaction
+
+
+class TestLogoutWithoutCASAuth(FunctionalTestCase):
```
Comment (phgross) [inline opengever/base/tests/test_logout_overlay.py]: Always use IntegrationTestCase's if you write new tests

File: opengever/core/upgrades/20180711170915_remove_logout_overlay_js/actions.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<object xmlns:i18n="http://xml.zope.org/namespaces/i18n" name="portal_actions" meta_type="Plone Actions Tool">
+
+  <!-- USER -->
+  <object name="user" meta_type="CMF Action Category" purge="False">
+
+    <object name="logout" meta_type="CMF Action" i18n:domain="plone">
+      <property name="title" i18n:translate="">Log out</property>
```
Comment (phgross) [inline opengever/core/upgrades/20180711170915_remove_logout_overlay_js/actions.xml]: Please add only the properties which has changed to the upgradestep xml.

File: opengever/core/upgrades/20180711170915_remove_logout_overlay_js/upgrade.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+from ftw.upgrade import UpgradeStep
+
+
+class RemoveLogoutOverlayJs(UpgradeStep):
```
Comment (phgross) [inline opengever/core/upgrades/20180711170915_remove_logout_overlay_js/upgrade.py]: `RemoveLogoutOverlayJs` is not 100% true --> `RemoveLogoutOverlay`

**Comment by tarnap** [comment]
@phgross: I still need to test if the LogoutWorker.js and LogoutClient.js still work.

File: opengever/core/upgrades/20180711170915_remove_logout_overlay/jsregistry.xml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<object name="portal_javascripts" meta_type="JavaScripts Registry">
+  <javascript id="@@logout_overlay.js" remove="True"/>
+
+  <javascript
```
Comment (phgross) [inline opengever/core/upgrades/20180711170915_remove_logout_overlay/jsregistry.xml]: The removal of the logout_overlay.js is missing in the upgradestep ...

Comment (tarnap) [reply opengever/core/upgrades/20180711170915_remove_logout_overlay/jsregistry.xml]: are you sure? https://github.com/4teamwork/opengever.core/pull/4578#discussion-diff-205779675R3

Comment (tarnap) [reply opengever/core/upgrades/20180711170915_remove_logout_overlay/jsregistry.xml]: screen shot 2018-07-27 at 16 51 04

Comment (phgross) [reply opengever/core/upgrades/20180711170915_remove_logout_overlay/jsregistry.xml]: ah sorry, did not see that. But the id is wrong: `++resource++opengever.document/logout_overlay.js`

**Pull request merged**